press/agent.py

```diff
@@ -101,6 +103,21 @@ def reinstall_site(self, site):
 			site=site.name,
 		)
 
+	def check_database_file(self, site):
+		remote_database_file = site.remote_database_file
+		mem_database_file = requests.get(url = frappe.get_doc("Remote File", site.remote_database_file).download_link)
+		# get frappe version from the first line
+		frappe_version = re.search("Frappe (\d{1,2})\.\d{1,2}\.\d{1,2}", gzip.decompress(mem_database_file.content).decode("utf-8"))
+
+		if not frappe_version:
+			raise Exception("The backup file is corrupted!")
+
+		source = frappe.get_doc("Bench", site.bench).apps[0].source
+		target_frappe_version = frappe.get_doc("App Source", source).versions[0].version
+
+		if frappe_version != target_frappe_version:
+			raise Exception("You're trying to restore site in the wrong version!")
```

Comment:
- Only reverse restores are not possible, forward restores should be fine (as they're followed with migrate. 
- App sources store `Version 14`, `Version 13` etc not exact version string which is present in backup headers.
- This should be warning an not completely blocking ?